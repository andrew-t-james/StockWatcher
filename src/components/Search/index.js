import React, { Component } from 'react';

class Search extends Component {
  constructor() {
    super();

    this.state = {
      userInput: '',
    };
  }

  handleInputChange(e) {
    const { value } = e.target;

    this.setState({
      userInput: value,
    });
  }

  handleSubmit(e) {
    e.preventDefault();
    console.log('working'); // TODO call fetch method here
    e.target.reset();
    this.setState({ userInput: '' });
  }

  render() {
    const { userInput } = this.state;
    return (
      <form onSubmit={e => this.handleSubmit(e)}>
        <input
          type="text"
          onChange={e => this.handleInputChange(e)}
          value={userInput}
          placeholder="Please Enter a symbol..."
        />
        <button>Add</button>
      </form>
    );
  }
}

export default Search;
