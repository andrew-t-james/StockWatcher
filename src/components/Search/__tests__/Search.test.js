import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Search from '../index';

describe('Search Unit tests', () => {
  let wrapper;

  beforeEach(() => (wrapper = shallow(<Search />)));
  afterEach(() => wrapper.unmount());

  test('should ', () => {
    const input = wrapper.find('input').length;
    const button = wrapper.find('button').length;

    expect(input).toBe(1);
    expect(button).toBe(1);
  });

  test('should should match snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  describe('text area updates', () => {
    beforeEach(() => {
      wrapper.find('input').simulate('change', {
        target: {
          value: 'new comment',
        },
      });
      wrapper.update();
    });

    test('updates on change and handles submit', () => {
      const actual = wrapper.find('input').prop('value');
      const expected = 'new comment';

      expect(actual).toEqual(expected);
    });

    test('when form is submitted text is cleared', () => {
      wrapper.find('form').simulate('submit', {
        preventDefault: jest.fn(),
        target: {
          reset: jest.fn(),
        },
      });

      wrapper.update();
      const actual = wrapper.find('input').prop('value');
      const expected = '';
      expect(actual).toEqual(expected);
    });
  });
});
