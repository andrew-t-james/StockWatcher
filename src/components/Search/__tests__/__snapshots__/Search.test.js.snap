// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Unit tests should should match snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    className="search__input"
    onChange={[Function]}
    placeholder="Enter stock symbol..."
    type="text"
    value=""
  />
  <button
    className="search__button"
  >
    Add
  </button>
</form>
`;
